mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java

```diff
@@ -293,4 +295,30 @@ public static boolean isInfrastructure(ParameterModel parameter) {
     return parameter.getModelProperty(InfrastructureParameterModelProperty.class).isPresent();
   }
 
+  /**
+   * Returns the first item in the {@code models} {@link List} that can be used implicitly.
+   * <p>
+   * A {@link ParameterizedModel} is consider to be implicit when all its {@link ParameterModel}s are optional
+   *
+   * @param models a {@link List} of {@code T}
+   * @param <T> the generic type of the items in the {@code models}. It's a type which is assignable from
+   *        {@link ParameterizedModel}
+   * @return one of the items in {@code models} or {@code null} if none of the models are implicit
+   */
+  public static <T extends ParameterizedModel> T getFirstImplicit(List<T> models) {
+    return models.stream()
+        .filter(ExtensionModelUtils::canBeUsedImplicitly)
+        .findFirst()
+        .orElse(null);
+  }
+
+  /**
+   * Returns whether the given model can be used implicitly. That means that all of its parameters are optional.
+   *
+   * @param parameterizedModel model
+   * @return whether the given model can be used implicitly or not.
+   */
+  public static boolean canBeUsedImplicitly(ParameterizedModel parameterizedModel) {
```

Comment:
This shold be a component model. There are (and there will be new) parameterized models for which this concept doesn't apply